graph-db/serverless/src/indexCmrCollection/__test__/handler.test.js

```diff
@@ -2,64 +2,61 @@ import nock from 'nock'
 
 import indexCmrCollection from '../handler'
 
-import { verifyGraphDb } from '../../testUtil/verifyGraphDb'
+import { updateCollection, deleteCollection } from '../../testUtil/indexCollection'
+
+import { verifyExistInGraphDb, verifyNotExistInGraphDb } from '../../testUtil/verifyGraphDb'
 
 beforeEach(() => {
   jest.clearAllMocks()
 })
 
+const getEvent = (conceptId, actionType) => {
+  const eventBody = `{"concept-id": "${conceptId}", "revision-id": "1", "action": "${actionType}" }`
+  return { Records: [{ body: eventBody }] }
+}
+
 describe('indexCmrCollection handler', () => {
   test('test index of single collection', async () => {
+    const datasetTitle = 'Latent reserves within the Swiss NFI'
+    const docName = 'https://en.wikipedia.org/wiki/latent_nfi'
+
+    await updateCollection('C1237293909-TESTPROV', datasetTitle, [docName])
+    await verifyExistInGraphDb(datasetTitle, docName)
+  })
+
+  test('test index of not found collection', async () => {
     nock(/local-cmr/)
-      .get(/search/)
-      .reply(200,
-        {
-          hits: 16996,
-          took: 5,
-          items: [{
-            meta: {
-              'concept-id': 'C1237293909-TESTPROV',
-              'provider-id': 'TESTPROV'
-            },
-            umm: {
-              RelatedUrls: [
-                {
-                  URLContentType: 'PublicationURL',
-                  Type: 'VIEW RELATED INFORMATION',
-                  Subtype: 'GENERAL DOCUMENTATION',
-                  URL: 'https://en.wikipedia.org/wiki/Buffalo_buffalo_Buffalo_buffalo_buffalo_buffalo_Buffalo_buffalo'
-                }
-              ],
-              DOI: {
-                DOI: 'doi:10.16904/envidat.166'
-              },
-              ShortName: 'latent-reserves-in-the-swiss-nfi',
-              EntryTitle: '\'Latent reserves\' within the Swiss NFI'
-            }
-          }]
-        })
-
-    const event = { Records: [{ body: '{"concept-id": "C1237293909-TESTPROV", "action": "concept-update" }' }] }
+      .get(/collections/)
+      .reply(200, {
+        hits: 0,
+        took: 6,
+        items: []
+      })
+
+    const event = getEvent('C123755555-TESTPROV', 'concept-update')
+
+    const consoleMock = jest.spyOn(console, 'log')
 
     const indexed = await indexCmrCollection(event)
 
+    expect(consoleMock).toBeCalledTimes(1)
+    expect(consoleMock).toBeCalledWith('Skip indexing of collection [C123755555-TESTPROV] as it is not found in CMR')
+
     const { body, statusCode } = indexed
 
-    expect(body).toBe('Successfully indexed 1 collection(s)')
+    expect(body).toBe('Successfully indexed 0 collection(s)')
```

Comment:
A better message for this example might be `Successfully indexed 0 collection(s). Skipped 1 collection(s)`. 

In `indexCmrCollection/handler.js` you could have a separate counter `skippedCount` that you only increment when `items.length === 0`, next to that console.log on line 48